### PR TITLE
Add React entities infinite scrolling

### PR DESCRIPTION
**Pull request by chrisdns**
- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/react/src/main/webapp/app/shared/util/url-utils.ts.ejs [hunk 1]
```diff
@@ -16,6 +16,40 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -%>
+export const parse = (header: string): any => {
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/shared/util/url-utils.ts.ejs]: can we use a lib instead doing it ourself, something like https://github.com/CassetteRocks/react-infinite-scroller can you please explore

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs [hunk 2]
```diff
@@ -139,7 +174,109 @@ export class <%= entityReactName %> extends React.Component<I<%= entityReactName
...
   render() {
-    const { <%=entityInstancePlural %>, match<% if (pagination !== 'no') { %>, totalItems<% } %>} = this.props;
+    const { <%=entityInstancePlural %>, match<% if (pagination === 'pagination' || pagination === 'pager') { %>, totalItems<% } %>} = this.props;
+    <%_ if (pagination === 'infinite-scroll') { _%>
+    const items = [];
+
+
+    <%=entityInstancePlural %>.map((<%=entityInstance %>, i) => (
+      items.push(
+        <tr key={`entity-${i}`}>
```
[the remark this answers is not in the document]
Comment (chrisdns) [reply generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs]: sorry for this I did not pay attention, i am going to fix it

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity.reducer.ts.ejs [hunk 3]
```diff
@@ -104,16 +115,42 @@ export default (state = initialState, action) => {
         ...state,
         loading: false,
         entities: action.payload.data
-    };
+      };
     <%_ } _%>
     case SUCCESS(ACTION_TYPES.FETCH_<%= entityActionNamePlural %>):
+      <%_ if (pagination === 'infinite-scroll') { _%>
+      const link = parse(action.payload.headers.link);
```
[the remark this answers is not in the document]
Comment (deepu105) [reply generators/entity-client/templates/react/src/main/webapp/app/entities/entity.reducer.ts.ejs]: can you please action on this as well? I guess you missed it

[the remark this answers is not in the document]
Comment (deepu105) [reply generators/entity-client/templates/react/src/main/webapp/app/entities/entity.reducer.ts.ejs]: meant for below logic

**Comment by deepu105** [comment]
I have released the lib, please update the PR

**Pull request merged**